src/Psl/Option/Option.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Option;
+
+use Closure;
+
+/**
+ * @template T
+ */
+final class Option
+{
+    /**
+     * @param ?array{T} $option
+     *
+     * @internal
+     */
+    private function __construct(
+        private readonly null|array $option,
```

Comment:
using `array{T}` instead of `T` to allow `null` as a valid value for `Option::some(null)`.

Reply:
Interesting solution! Alternatively, it could be just an additional boolean flag indicating the value is none or some, but type-wise your solution is better I think.

With boolean flag there could be problems like here: https://github.com/marcosh/lamphpda/blob/master/src/Maybe.php#L118

Reply:
yes, this is a type-safe way of doing it, initially, it was `array{false}|array{true, T}`, but psalm doesn't really like that.